pkg/agent/agent_windows.go

```diff
@@ -0,0 +1,138 @@
+	if err := i.ofClient.InstallExternalFlows(nodeIP, *subnetCIDR); err != nil {
+		klog.Errorf("Failed to setup SNAT openflow entries: %v", err)
+		return err
+	}
+	return nil
+}
+
+// prepareHostNetworking creates HNS Network for containers.
+func (i *Initializer) prepareHostNetworking() error {
+	return util.PrepareHostNetworking(i.nodeConfig.PodCIDR, i.nodeConfig.NodeIPAddr)
+}
+
+// prepareOVSBridge adds local port and uplink to ovs bridge.
+func (i *Initializer) prepareOVSBridge() error {
+	hnsNetwork, err := hcsshim.GetHNSNetworkByName(util.LocalHNSNetwork)
+	if err != nil {
+		return err
+	}
+
+	// Set datapathID of OVS bridge.
+	// If no datapathID configured explicitly, the reconfiguration operation will change OVS bridge datapathID
+	// and break the OpenFlow channel.
+	datapathID := strings.Replace(hnsNetwork.SourceMac, ":", "", -1)
+	datapathID = "0000" + datapathID
+	if err = i.ovsBridgeClient.SetDatapathID(datapathID); err != nil {
+		klog.Errorf("Failed to set datapath_id %s: %v", datapathID, err)
+		return err
+	}
+
+	// Create local port.
+	brName := i.ovsBridgeClient.GetBridgeName()
+	if _, err := i.ovsBridgeClient.GetOFPort(brName); err == nil {
+		klog.Infof("OVS bridge local port %s already exists, skip the configuration", brName)
+	} else {
+		if _, err := i.ovsBridgeClient.CreateInternalPort(brName, config.AutoAssignedOFPort, nil); err != nil {
+			return err
+		}
+	}
+
+	// If uplink is already exists, return.
+	uplink := hnsNetwork.NetworkAdapterName
+	i.nodeConfig.UplinkName = uplink
+	if _, err := i.ovsBridgeClient.GetOFPort(uplink); err == nil {
+		klog.Infof("Uplink %s already exists, skip the configuration", uplink)
+		return err
+	}
+	// Create uplink port.
+	uplinkPortUUId, err := i.ovsBridgeClient.CreateUplinkPort(uplink, config.UplinkOFPort, nil)
+	if err != nil {
+		klog.Errorf("Failed to add uplink port %s: %v", uplink, err)
+		return err
+	}
+	uplinkInterface := interfacestore.NewUplinkInterface(uplink)
+	uplinkInterface.OVSPortConfig = &interfacestore.OVSPortConfig{uplinkPortUUId, config.UplinkOFPort}
+	i.ifaceStore.AddInterface(uplinkInterface)
+
+	// Get IP, MAC of uplink interface
+	ipAddr, ipNet, err := net.ParseCIDR(hnsNetwork.ManagementIP)
+	if err != nil {
+		klog.Errorf("Failed to parse IP Address %s for HNSNetwork %s: %v",
+			hnsNetwork.ManagementIP, util.LocalHNSNetwork, err)
+		return err
+	}
+	ifIpAddr := net.IPNet{IP: ipAddr, Mask: ipNet.Mask}
+	klog.Infof("Found hns network management ipAddr: %s", ifIpAddr.String())
+	// Move IP, MAC of from uplink interface to OVS bridge
+	err = util.EnableHostInterface(brName)
+	if err != nil {
+		return err
+	}
+	macAddr, err := net.ParseMAC(hnsNetwork.SourceMac)
+	if err != nil {
+		return err
+	}
+	err = util.ConfigureMacAddress(brName, macAddr)
+	if err != nil {
+		klog.Errorf("Failed to set Mac Address %s for interface %v: ", macAddr, uplink, err)
+		return err
+	}
+	existingIpAddr, err := util.GetAdapterIPv4Addr(brName)
```

Comment:
Looks you forget to move the default gateway on uplink interface to bridge interface?

Reply:
Done